### PR TITLE
Handle empty items in arrays and arraylike objects

### LINKED ISSUE #74: Sparse arrays are treated like arrays of undefined
`concordance` treats empty slots in arrays like `undefined`, for both formatting and comparison.

`format(new Array(5))` gives
```
[
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
]
```

whereas `util.inspect(new Array(5))` gives `[ <5 empty items> ]`

`compare(new Array(5), Array.from({length: 5})` returns `true`, although these values behave rather differently.

### PR DESCRIPTION
**Pull request by ninevra**
Adds an `EmptyItem` descriptor to represent the gaps in sparse arrays. Formats gaps as `<empty item>`, and treats them as unequal to anything other than empty items for comparison and diffing.

Adds theme keys

```js
item: {
    empty: {
        open: '<',
        close: '>',
    }
}
````

for controlling the formatting; I'm not sure if this is the best way to do this. Maybe the item text should be configurable, or maybe the brackets should be part of the item text and not part of the open/close wrappers?

I initially implemented this with an `Empty` primitive descriptor, but I think `EmptyItem` better matches the data model. I can't think of a non-list context where `Empty` would be valid.

This PR does not collapse empty items into e.g. `<5 empty items>` like `util.inspect` does; that functionality seemed like a subset of #12.

Closes #74.

This is a breaking change & bumps the serialization format version to 4.

### CODE REVIEW

**Comment by ninevra** [comment]
There's a bug in the alignment code that I need to figure out.

File: test/compare.js [hunk 1]
```diff
@@ -66,3 +66,11 @@ test('arrays are also compared by property', t => {
...
   t.false(compare(a3, a4).pass)
 })
+
+test('empty array slots do not equal undefined', t => {
+  t.false(compare(new Array(1), Array.from({ length: 1 })).pass)
+})
+
+test('empty arraylike slots do not equal undefined', t => {
+  t.false(compare({ length: 2, 0: 'a' }, { length: 2, 0: 'a', 1: undefined }).pass)
+})
```
Comment (novemberborn) [inline test/compare.js:76]: Perhaps this needs some tests for equality between arrays with empty items?

**Review by novemberborn — COMMENTED** [review]
This is great!

> Maybe the item text should be configurable, or maybe the brackets should be part of the item text and not part of the open/close wrappers?

Perhaps the entire `<empty item>` could be the theme string, and then tools like AVA can color it however they like?

> I initially implemented this with an `Empty` primitive descriptor, but I think `EmptyItem` better matches the data model. I can't think of a non-list context where `Empty` would be valid.

👍 

> This PR does not collapse empty items into e.g. `<5 empty items>` like `util.inspect` does; that functionality seemed like a subset of #12.

👍 

> This is a breaking change & bumps the serialization format version to 4.

👍
